### DISCUSSION TITLE
Error when getting data from async function after reloading

### DISCUSSION THREAD

**Discussion by priverop** [nuxt/nuxt#20000] [Questions]
Hello,

If I enter a page through a NuxtLink everything works fine, but if I reload the page after or enter the URL manually, it gives me this error:

```
[Vue warn]: Unhandled error during execution of setup function at <PageName onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > >
```

After debugging I found the issue is here:
```javascript
const { data: user } = await getUser(uid)
console.log(user.value)
```

This getUser function returns a Promise:
```typescript
/**
* Get make
* @returns {Promise<any>}
* @throws {Error} Error name
**/
export const getUser = (uid: any): Promise<any> => {
  return apiFetch(`/api/users/${uid}`, { method: 'GET' })
}
```
(apiFetch is a wrapper of useFetch)

When I enter the page with the NuxtLink this is the result of the console.log:

![Captura de pantalla 2023-03-30 a las 13 06 56](https://user-images.githubusercontent.com/1587633/228837681-d7342e16-dc09-4d99-87e1-2724b04fdc37.png)

After that if I reload the page I get "undefined" in that console.log

**Comment by ndragun92** [comment]
Would be good if you create minimal reproduction on StackBlitz or somewhere else
